php-classes/Media.class.php

```diff
@@ -463,6 +463,12 @@ public static function createFromUpload($uploadedFile, $fieldValues = array())
     public static function createFromFile($file, $fieldValues = array())
     {
         try {
+            // check if url
+            if (filter_var($file, FILTER_VALIDATE_URL)) { 
+                copy($file, '/tmp/tmp-image.png');
+                $file = '/tmp/tmp-image.png';
```

Comment:
@jtwiest use [tempnam](http://php.net/manual/en/function.tempnam.php) to generate an actually unique filename so that parallel processes won't step on each other.

Store a copy of this temporary filename in another variable (e.g. `$file = $tempFile = tempnam('/tmp', 'remote_media')`) and then use a `finally` block to ensure that `$tempFile` gets deleted if it isset